### PR TITLE
change LayoutLMForSequenceClassification inference accuracy tolerance

### LINKED ISSUE #128510: [inductor][cpu]LayoutLMForSequenceClassification AMP single threadstatic/dynamic shape default/CPP wrapper accuracy failure
### 🐛 Describe the bug

ffc202a1b91def8c81a6eb9a39777bc7e149e1ee
```
/workspace/pytorch# bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first static default
Testing with freezing on.
single-thread testing....
/opt/conda/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  LayoutLMForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
E0612 14:35:01.137710 128914524942784 torch/_dynamo/utils.py:1476] RMSE (res-fp64): 0.00157, (ref-fp64): 0.00046 and shape=torch.Size([1, 2]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000
E0612 14:35:01.137950 128914524942784 torch/_dynamo/utils.py:1390] Accuracy failed for key name logits
fail_accuracy
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips
cpu,LayoutLMForSequenceClassification,1,fail_accuracy,396,3,2,1,0,0,2
```
c4468518293f524d7ff1b2514d59753870dc232c
```
/workspace/pytorch# bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first static default
Testing with freezing on.
single-thread testing....
/opt/conda/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  LayoutLMForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
pass
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips
cpu,LayoutLMForSequenceClassification,1,pass,396,3,2,1,0,0,2
```

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>name</th>
      <th>target_branch</th>
      <th>target_commit</th>
      <th>refer_branch</th>
      <th>refer_commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>torchbench</td>
      <td>main</td>
      <td>d6015d42</td>
      <td>main</td>
      <td>d6015d42</td>
    </tr>
    <tr>
      <td>torch</td>
      <td>main</td>
      <td>f681e3689b857b8811f19d60d439bfb3fb2dd2d3</td>
      <td>main</td>
      <td>c19ad112f65ab88af3eeb319dda0675c076b0b67</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>main</td>
      <td>2.2.0a0+1980f8a</td>
      <td>main</td>
      <td>2.2.0a0+1980f8a</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>main</td>
      <td>nightly</td>
      <td>main</td>
      <td>nightly</td>
    </tr>
  </tbody>
</table>

</table>

Repro:
[inductor_single_run.sh](https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/inductor_single_run.sh)
bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first dynamic/static default/cpp
Suspected guilty commit: https://github.com/pytorch/pytorch/commit/ffc202a1b91def8c81a6eb9a39777bc7e149e1ee
[huggingface-LayoutLMForSequenceClassification-inference-amp-static-default-single-accuracy-crash_guilty_commit.log](https://github.com/user-attachments/files/15806944/huggingface-LayoutLMForSequenceClassification-inference-amp-static-default-single-accuracy-crash_guilty_commit.log)
cc @ezyang @anijain2305 @chauhang @WeizhuoZhang-intel @chuanqi129

### PR DESCRIPTION
**Pull request by Valentine233**
Fixes #128510.

https://github.com/pytorch/pytorch/pull/124451 makes LayoutLMForSequenceClassification hit the SDPA pattern 1 and then encounter the accuracy issue. The issue only happens with BF16 inference single thread. This PR tends to increase the model tolerance and make the check pass. Note that even the math-version SDPA could have the issue because of some small implementation diff.

The test log:
Single thread
```
correct_result:  SequenceClassifierOutput(loss=tensor(0.5998), logits=tensor([[0.3301, 0.1338]], dtype=torch.bfloat16), hidden_states=None, attentions=None)
new_result:  SequenceClassifierOutput(loss=tensor(0.6016), logits=tensor([[0.3281, 0.1357]], dtype=torch.bfloat16), hidden_states=None, attentions=None)
E0627 01:09:16.762789 140281313759104 torch/_dynamo/utils.py:1476] RMSE (res-fp64): 0.00151, (ref-fp64): 0.00046 and shape=torch.Size([1, 2]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000
E0627 01:09:16.762972 140281313759104 torch/_dynamo/utils.py:1390] Accuracy failed for key name logits
fail_accuracy
```

Multiple threads
```
correct_result:  SequenceClassifierOutput(loss=tensor(0.6007), logits=tensor([[0.3301, 0.1357]], dtype=torch.bfloat16), hidden_states=None, attentions=None)
new_result:  SequenceClassifierOutput(loss=tensor(0.6016), logits=tensor([[0.3281, 0.1357]], dtype=torch.bfloat16), hidden_states=None, attentions=None)
pass
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Review by eellison — COMMENTED** [review]
can we only bump tolerance on cpu?

**Comment by Valentine233** [comment]
> can we only bump tolerance on cpu?

Changed to cpu only.

**Review by jansel — APPROVED** [review]
